WordPress/src/main/java/org/wordpress/android/reader/savedposts/provider/ReaderSavedPostsProvider.kt

```diff
@@ -0,0 +1,60 @@
+package org.wordpress.android.reader.savedposts.provider
+
+import android.database.Cursor
+import android.net.Uri
+import org.wordpress.android.WordPress
+import org.wordpress.android.datasets.wrappers.ReaderPostTableWrapper
+import org.wordpress.android.models.ReaderTag
+import org.wordpress.android.models.ReaderTagType.BOOKMARKED
+import org.wordpress.android.provider.query.QueryContentProvider
+import org.wordpress.android.provider.query.QueryResult
+import org.wordpress.android.util.publicdata.JetpackPublicData
+import org.wordpress.android.util.signature.SignatureNotFoundException
+import org.wordpress.android.util.signature.SignatureUtils
+import javax.inject.Inject
+
+class ReaderSavedPostsProvider  : QueryContentProvider() {
+    @Inject lateinit var queryResult: QueryResult
+    @Inject lateinit var readerPostTableWrapper: ReaderPostTableWrapper
+    @Inject lateinit var signatureUtils: SignatureUtils
```

Comment:
I've extracted the common Provider logic to `ClientVerification` class (example [here](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/sharedlogin/provider/SharedLoginProvider.kt#L33)). Would it make sense to use it instead?

Basically it would replace lines 37-41 with something like:
```
if (clientVerification.canTrust(callingPackage)) {
```
And `SignatureUtils` dependency would be replaced by `ClientVerification`.

Reply:
I may have missed it after merging from trunk, makes total sense 👍